version: '3.7'

services:
  nginx:
    container_name: web_server
    restart: on-failure
    image: jonasal/nginx-certbot:latest
    env_file:
      - ./.env.prod.nginx
    volumes:
      - nginx_secrets:/etc/letsencrypt
      - ./nginx.conf:/etc/nginx/user_conf.d/nginx.conf
    ports:
      - "80:80"
      - "443:443"
    depends_on:
      - web
  web:
    build:
      context: .
      dockerfile: Dockerfile.prod
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - static_volume:/home/app/web/staticfiles
    ports:
      - "8000:8000"
    env_file:
      - ./.env.prod
    depends_on:
      - db
      - redis
  db:
    image: postgres:15
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    env_file:
      - ./.env.prod.db
  redis:
    image: redis:6.2-alpine
    container_name : redis
    restart: always
    ports:
      - '6379:6379'
    command: redis-server --save 20 1 --loglevel warning
    volumes:
      - redis:/data

volumes:
  postgres_data:
  static_volume:
  nginx_secrets:
  redis:
    driver: local
